File: tv_app_backend/models/UserLocation.js
const mongoose = require('mongoose');

const userLocationSchema = new mongoose.Schema({
  userId: {
    type: String,
    required: true,
  },
  province: {
    type: String,
    required: true,
  },
  city: {
    type: String,
    required: true,
  },
});

const UserLocation = mongoose.model('UserLocation', userLocationSchema);

module.exports = UserLocation; 

File: lib/models/user_location.dart
class UserLocation {
  final String userId;
  final String province;
  final String city;

  UserLocation({
    required this.userId,
    required this.province,
    required this.city,
  });

  factory UserLocation.fromJson(Map<String, dynamic> json) {
    return UserLocation(
      userId: json['user_id'].toString(),
      province: json['province'].toString(),
      city: json['city'].toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'user_id': userId,
      'province': province,
      'city': city,
    };
  }
}
